### DISCUSSION TITLE
param.query object becomes empty on server side.

### DISCUSSION THREAD

**Discussion by sachinsawant71** [feathersjs/feathers#2657] [Q&A] [answered]
### params.query object becomes empty on server side

I am using websocket for client server communication. 
I am trying to send query object for find service method call. I setup query to include criteria and call service.find.

On the server side, within the hook I see that the query object is empty. It does not contain the criteria I added when i called the service from the client. Some logic inside the framework is resetting it. 

Feathers version - 4.3.1
Node version - v14.17.3

**Comment by daffl** [comment]
The logic is more likely in your application code than the framework but we'd need a complete example to reproduce your issue.

**Comment by sachinsawant71** [comment]
Here are additional details -

- The application uses feathers.js in the backend 
- The front end is written in react.js. It uses feathersjs client for communication with the backend using websocket as the transport layer
- For user authentication, we use email/password i.e. local authentication. It is working as expected 
- Recently, I have added auth0/SSO as an additional authentication method. Here is the the logic, I have implemented, 
                        - > Frontend is configured to use auth0-spa-js client to communicate with Auth0 and get auth0 token
                        - > I have written a custom authentication strategy to handle the auth0 based login - the new class verifies the auth0 token and returns the response. The response contains local featherjs JWT token. It is used for all other api communications. 

Problem statement -

Auth0-custom-authentication strategy flow works fine. When I try to use generated feathers JWT token for API calls, i have observed that 

- API calls that do not use query work fine 
- When I call any api which has query object, the query object becomes empty ({}) at the server side. I debugged the issue using ws communication in chrome devtools. In the client server communication, client is properly sending query object to the server. I also observed that the server is also receiving query object properly (Using WS communication listener in the server).  When I check query in the "hook", it is empty. I also verified it in application hook i.e. very beginning of the communication in the feathersJS api communication. 

I need help to debug this issue as I am not sure why it is happening.

Another puzzling observation - If I just refresh the browser side application using "refresh" button, communication is restored. Everything starts working as expected including the query object.

**Reply by daffl** [reply]
Hm, this might be a problem in the authentication flow but an app to reproduce the problem would still be helpful.

**Reply by sachinsawant71** [reply]
I will get a sample app.

One more observation - get call with query parameter works fine. Only the find call with query parameter is creating challenges.

**Comment by sachinsawant71** [comment]
I have updated the feathersJS library from 4.3.1 to 4.5.15. It has resolved the query issue I was facing. So we can close the thread at this moment.

NOTE - My application is not working as expected after Auth0 SSO login. There are some service calls in which query object is still going empty. I want debug it at my end to ensure that it is not a code issue. 

QUESTION - I want to debug feathers framework code check if the query parameters are being passed to each hook. Where do I start for it? Can you guide me?

**Comment by sachinsawant71** [comment, marked as answer]
I am able to resolve the issue. It was little tricky. I have custom authentication scheme. The scheme was returning a result and the result had a field with name "query". This field was somehow overriding "query" parameters passed by service calls and causing the challenges I faced. 

I renamed the query field returned by my custom authentication scheme and everything started working as expected.
